ios/EmailShare.m

```diff
@@ -32,23 +63,109 @@ - (void)shareSingle:(NSDictionary *)options
         }
 
         message = [RCTConvert NSString:options[@"message"]];
-
+        
+        [mc setToRecipients:@[email]];
+        [mc setSubject:subject];
+        [mc setMessageBody:message isHTML:NO];
+        
         if ([options objectForKey:@"url"] && [options objectForKey:@"url"] != [NSNull null]) {
-            message = [message stringByAppendingString: [@" " stringByAppendingString: [RCTConvert NSString:options[@"url"]]] ];
-        }
-
-        NSString * urlWhats = [NSString stringWithFormat:@"mailto:%@?subject=%@&body=%@", email, subject, message ];
-        NSURL * whatsappURL = [NSURL URLWithString:[urlWhats stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
```

Comment:
can we keep this?

Reply:
They are not really equivalent, and it's either one or the other.  What would be the rationale to keep it?  We might need a new flag/setting to enable both options.

Reply:
Looking at it again, these lines are in fact redundant and can be achieved with the same `Linking` API that's already part of RN.  So it makes even more sense for the Share API to use an improved version to send emails.